src/program.js

```diff
@@ -174,6 +178,32 @@ export class Program {
     return argv;
   }
 
+  // Remove WEB_EXT_* environment vars that are not a global cli options
+  // or an option supported by the current command (See #793).
+  cleanupProcessEnvConfigs(systemProcess: typeof process) {
+    const cmd = yargsParser(this.programArgv)._[0];
+    const env = systemProcess.env || {};
+    const toOptionKey = (k) => decamelize(
+      camelCase(k.replace(`${envPrefix}_`, '')),
+      '-'
+    );
+
+    if (cmd) {
+      Object.keys(env)
+        .filter((k) => k.startsWith(envPrefix))
```

Comment:
You're looking for keys starting with `envPrefix`, but processing keys with the assumption that they start with `${envPrefix}_`. Let's be consistent.

```javascript
let prefix = `${envPrefix}_`;
...
      Object.keys(env)
        .filter((k) => k.startsWith(prefix))
        .map((k) => k.slice(prefix.length))
        .map((k) => decamelize(camelCase(k), '-'))
        .forEach((optKey) => {
```

Side note: In our documentation, we are only suggesting the use of `WEB_EXT_*`, but practice it seems that `WEB_EXT*` (without the second underscore) is also supported. Since we haven't documented that option, I'm fine with only considering the `WEB_EXT_*` case.

Reply:
> You're looking for keys starting with `envPrefix`, but processing keys with the assumption that they start with `${envPrefix}_`. Let's be consistent.

yeah, that's fair :+1:

> Side note: In our documentation, we are only suggesting the use of `WEB_EXT_*`, but practice it seems that `WEB_EXT*` (without the second underscore) is also supported. Since we haven't documented that option, I'm fine with only considering the `WEB_EXT_*` case.

uhm... let me check first what yargs "thinks" about the `WEB_EXT*`, I haven't checked if it is parsing those as well (but from your comment I guess it does, and so it may be more reasonable to make sure that they are not going to trigger the same issue if someone is using them "by mistake", e.g. because of a typo).